packages/finance/src/liability/mod.rs

```diff
@@ -91,18 +91,28 @@ impl Liability {
         self.recalc_time
     }
 
-    pub fn init_borrow_amount<P>(&self, downpayment: P) -> P
+    pub fn init_borrow_amount<P>(&self, downpayment: P, max_ltv: Option<NonZeroPercent>) -> P
     where
         P: Percentable,
     {
         use crate::fraction::Fraction;
+
+        debug_assert!(self.initial > Percent::ZERO);
         debug_assert!(self.initial < Percent::HUNDRED);
 
-        // borrow = init%.of(borrow + downpayment)
-        // (100% - init%).of(borrow) = init%.of(downpayment)
-        // borrow = init% / (100% - init%) * downpayment
-        let ratio = Rational::new(self.initial, Percent::HUNDRED - self.initial);
-        ratio.of(downpayment)
+        #[cfg(debug_assertions)]
+        if let Some(max_ltv) = max_ltv {
+            debug_assert!(max_ltv.percent() < Percent::HUNDRED);
+        }
```

Comment:
such debug assert makes sense on the value = min of the two ... it's more important to that... though self.initial is already check that is less than 100%